changelog/@unreleased/pr-630.v2.yml

```diff
@@ -0,0 +1,6 @@
+type: fix
+fix:
+  description: Structured errors are now correctly deserialized when requests exhaust
```

Comment:
For what it's worth I think this was only a bug for 503 and 429 responses where our infrastructure sends empty response bodies.